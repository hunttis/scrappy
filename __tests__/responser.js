const responser = require('../responser')

const previousCards = [
  {
    spoilBlock: 'UNSTABLE',
    spoilDate: 'November 16',
    cardUrl: 'ust/cards/druidofthesacredbeaker.jpg'
  },
  {
    spoilBlock: 'UNSTABLE',
    spoilDate: 'November 16',
    cardUrl: 'ust/cards/betterthanone.jpg'
  },
  {
    spoilBlock: 'UNSTABLE',
    spoilDate: 'November 16',
    cardUrl: 'ust/cards/hydradoodle.jpg'
  }
]

const newCards = [
  {
    spoilBlock: 'UNSTABLE',
    spoilDate: 'November 21',
    cardUrl: 'ust/cards/urzaacademyheadmaster.jpg'
  },
  {
    spoilBlock: 'UNSTABLE',
    spoilDate: 'November 16',
    cardUrl: 'ust/cards/drjuliusjumblemorph.jpg'
  },
  {
    spoilBlock: 'UNSTABLE',
    spoilDate: 'November 16',
    cardUrl: 'ust/cards/druidofthesacredbeaker.jpg'
  },
  {
    spoilBlock: 'UNSTABLE',
    spoilDate: 'November 16',
    cardUrl: 'ust/cards/betterthanone.jpg'
  },
  {
    spoilBlock: 'UNSTABLE',
    spoilDate: 'November 16',
    cardUrl: 'ust/cards/hydradoodle.jpg'
  },
  {
    spoilBlock: 'RIVALS OF IXALAN',
    spoilDate: 'November 16',
    cardUrl: 'ust/cards/superduperdeathray.jpg'
  }]

describe('responser', () => {
  describe('getNewCards()', () => {
    it('should return difference of two card lists', () => {
      const newCardList = responser.getNewCards(previousCards, newCards)
      expect(newCardList).toMatchSnapshot()
    })
  })
  describe('getResponse()', () => {
    it('should return slack formatted response', () => {
      const response = responser.getResponse(previousCards, newCards)
      expect(response).toMatchSnapshot()
    })
  })
})
