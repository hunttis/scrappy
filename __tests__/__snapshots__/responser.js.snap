// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responser getNewCards() should return difference of two card lists 1`] = `
Array [
  Object {
    "cardUrl": "ust/cards/urzaacademyheadmaster.jpg",
    "spoilBlock": "UNSTABLE",
    "spoilDate": "November 21",
  },
  Object {
    "cardUrl": "ust/cards/drjuliusjumblemorph.jpg",
    "spoilBlock": "UNSTABLE",
    "spoilDate": "November 16",
  },
  Object {
    "cardUrl": "ust/cards/superduperdeathray.jpg",
    "spoilBlock": "RIVALS OF IXALAN",
    "spoilDate": "November 16",
  },
]
`;

exports[`responser getResponse() should return slack formatted response 1`] = `
Object {
  "body": Object {
    "attachments": Array [
      Object {
        "footer": "UNSTABLE, RIVALS OF IXALAN",
      },
    ],
    "channel": "#mtg",
    "icon_emoji": ":mtg-black:",
    "text": "New spoilers! <http://mythicspoiler.com/newspoilers.html|Take a look!>",
    "username": "Scrappy",
  },
  "json": true,
  "method": "POST",
  "uri": "https://hooks.slack.com/services/123",
}
`;
